content/blog/kubernetes-clusters-as-code-part1.md

```diff
@@ -0,0 +1,922 @@
+      version = ">= 0.2.2"
+    }
+  }
+}
+ 
+provider hpegl {
+  caas {
+  }
+}
+```
+
+<!--StartFragment-->
+
+<br/>
+
+## Create a Cluster resource
+
+### Terraform data source for Cluster Blueprint
+
+In order to use the data source available for cluster blueprint, you should add the below block in your terraform file, and specify the cluster blueprint name. Using this data source, terraform will fetch the cluster Blueprint ID associated with it.
+
+In the below block, "demo" is the cluster blueprint **name** provided:
+
+<!--EndFragment-->
+
+```markdown
+data "hpegl_caas_cluster_blueprint" "bp" {
+  name = "demo"
+  site_id = data.hpegl_caas_site.blr.id
+}
+```
+
+<!--StartFragment-->
+
+The cluster blueprint ID can now be fetched by using:
+
+<!--EndFragment-->
+
+```markdown
+blueprint_id = data.hpegl_caas_cluster_blueprint.bp.id
+```
+
+<!--StartFragment-->
+
+<br/>
+
+### Terraform data source for Site
+
+In order to use the data source available for site, you should add the below block in your terraform file, and specify the site name. Using this data source, Terraform will fetch the site ID associated with it.
+
+In the below block, "BLR" is the site **name** provided:
+
+<!--EndFragment-->
+
+```markdown
+data "hpegl_caas_site" "blr" {
+  name = "BLR"
+  space_id = ""
+}
+```
+
+<!--StartFragment-->
+
+The site ID can now be fetched by using:
+
+<!--EndFragment-->
+
+```markdown
+site_id = data.hpegl_caas_site.blr.id
+```
+
+<!--StartFragment-->
+
+<br/>
+
+###  Terraform data source for Cluster
+
+In order to use the data source available for cluster, you should add the below block and provide the cluster name and space id. Using this data source, terraform will fetch the cluster server and user token associated with it.
+
+In the below block, "tf-test" is the **name** of the pre-created cluster.
+
+<!--EndFragment-->
+
+```markdown
+data "hpegl_caas_cluster" "test" {
+  name     = "tf-test"
+  space_id = ""
+}
+```
+
+<!--StartFragment-->
+
+The **host** (cluster server) and **token** (user token) can now be fetched from the cluster kubeconfig using:
+
+<!--EndFragment-->
+
+```markdown
+host     = yamldecode(base64decode(data.hpegl_caas_cluster.test.kubeconfig)).clusters[0].cluster.server
+token    = yamldecode(base64decode(data.hpegl_caas_cluster.test.kubeconfig)).users[0].user.token
+```
+
+<!--StartFragment-->
+
+<br/>
+
+### Terraform resource for Cluster
+
+In order to create a cluster using the cluster resource, the following values should be specified in the **cluster-create.tf** file:
+
+1. Site Name: Fill in the appropriate site **name** in the **hpegl_caas_site** block. In the below example, name= "BLR" 
+2. Cluster Blueprint Name: Fill in the appropriate cluster blueprint **name** in the **hpegl_caas_cluster_blueprint** block. In the below example, name= "demo" 
+3. Cluster Name: Fill in the cluster **name** of your choice in the **hpegl_caas_cluster** block. In the below example, name= "tf-test"
+
+```markdown
```

Comment:
Suggestion to replace the code snippet for the Note with the markdown syntax below:

\>Note: Here, the space_id is automatically set to the value specified while exporting TF\_VAR\_HPEGL\_SPACE.

Reply:
Done